src/activitys/ChallengeModeDungeon.ts

```diff
@@ -0,0 +1,158 @@
+import { Metadata } from "main/types";
+import { dungeonTimersByMapId, VideoCategory } from "../main/constants";
+import { ChallengeModeTimelineSegment, TimelineSegmentType } from "../main/keystone";
+import Activity from "./Activity";
+
+export default class ChallengeModeDungeon extends Activity {
+    private mapID: number;
+    private level: number;
+    private affixes: number[];
+    private timings: number[];
```

Comment:
We don't actually need this anymore; we need to just remove that and calculate the upgradeLevel and save that with the metadata - that way all that logic I made some time ago regarding this will be moot and could be removed.

You don't have to do that now, though, I can do that at a later time when you're done with this.